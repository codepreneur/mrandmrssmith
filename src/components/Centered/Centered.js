import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'

import { withStyles } from '@material-ui/core/styles'

const styles = {
  root: {
    alignItems: 'center',
    alignSelf: 'stretch',
    display: 'flex',
    flexGrow: 1,
    justifyContent: 'center',
    overflowY: 'scroll',
    paddingTop: '4rem',
    paddingBottom: '4rem',
  },
}

const Centered = ({
  component: Component,
  classes,
  className,
  children,
  ...forwardedProps
}) => (
  <Component {...forwardedProps} className={clsx(classes.root, className)}>
    {children}
  </Component>
)

Centered.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  classes: PropTypes.object.isRequired,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
}

Centered.defaultProps = {
  component: 'div',
}

export default withStyles(styles)(Centered)
