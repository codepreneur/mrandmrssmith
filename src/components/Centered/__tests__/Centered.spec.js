import React from 'react'
import { shallow } from 'enzyme'

import Centered from '../Centered'

describe('Centered component', () => {
  describe('smoke test', () => {
    it('should render without any errors', () => {
      expect(() => {
        shallow(<Centered />)
      }).not.toThrow()
    })

    it('should render correctly', () => {
      const wrapper = shallow(<Centered />)
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.dive()).toMatchSnapshot()
    })
  })

  describe('children prop test', () => {
    it('should not have children', () => {
      expect(shallow(<Centered />).children()).toHaveLength(0)
    })

    it('should have children', () => {
      const Content = () => <div>content</div>
      const wrapper = shallow(
        <Centered>
          <Content />
        </Centered>,
      )
      expect(wrapper.children()).toHaveLength(1)
      expect(wrapper.children().contains(<Content />)).toBe(true)
    })
  })

  describe('component prop test', () => {
    it('should have correct default component type', () => {
      expect(
        shallow(<Centered />)
          .dive()
          .type(),
      ).toEqual('div')
    })

    it('should custom component type', () => {
      const Custom = () => null
      expect(
        shallow(<Centered component="section" />)
          .dive()
          .type(),
      ).toEqual('section')
      expect(
        shallow(<Centered component={Custom} />)
          .dive()
          .type(),
      ).toBe(Custom)
    })
  })

  describe('className prop test', () => {
    it('should contain default className', () => {
      expect(
        shallow(<Centered />)
          .dive()
          .hasClass('root'),
      ).toBe(true)
    })

    it('should contain custom className', () => {
      const wrapper = shallow(<Centered className="customClass" />)
      expect(wrapper.dive().hasClass('root')).toBe(true)
      expect(wrapper.dive().hasClass('customClass')).toBe(true)
    })
  })

  describe('props forwarding test', () => {
    it('should forward other props', () => {
      const wrapper = shallow(<Centered custom1="foo" custom2 />)
      expect(wrapper.dive().prop('custom1')).toBe('foo')
      expect(wrapper.dive().prop('custom2')).toBe(true)
    })

    it('should not forward own props', () => {
      const wrapper = shallow(<Centered component="article" />)
      expect(wrapper.dive().prop('children')).toBeUndefined()
      expect(wrapper.dive().prop('classes')).toBeUndefined()
      expect(wrapper.dive().prop('component')).toBeUndefined()
    })
  })
})
