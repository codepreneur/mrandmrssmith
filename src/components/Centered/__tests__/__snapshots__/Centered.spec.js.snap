// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered component smoke test should render correctly 1`] = `
<Centered
  classes={null}
  component="div"
/>
`;

exports[`Centered component smoke test should render correctly 2`] = `
<div
  className="root"
/>
`;
